sdk/openai/openai-assistants/test/public/utils/utils.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  PipelineRequest,
+  PipelineResponse,
+  createDefaultHttpClient,
+  createEmptyPipeline,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { randomUUID } from "@azure/core-util";
+import { KeyCredential } from "@azure/core-auth";
+import { AuthMethod } from "./recordedClient.js";
+import {
+  Recorder,
+  assertEnvironmentVariable,
+  isLiveMode,
+  isRecordMode,
+} from "@azure-tools/test-recorder";
+import { OpenAIKeyCredential } from "../../../src/index.js";
+
+function toString(error: any): string {
+  return error instanceof Error ? error.toString() + "\n" + error.stack : JSON.stringify(error);
+}
+
+export async function withDeployments<T>(
+  deployments: string[],
+  run: (model: string) => Promise<T>,
+  validate: (result: T) => void
+): Promise<string[]> {
+  const errors = [];
+  const succeeded = [];
+  assert.isNotEmpty(deployments, "No deployments found");
+  for (const deployment of deployments) {
+    try {
+      console.log(`testing with ${deployment}`);
+      const res = await run(deployment);
+      if (!isRecordMode()) {
+        validate(res);
+      }
+      succeeded.push(deployment);
+    } catch (e) {
+      const error = e as any;
+      if (!e) continue;
+      const errorStr = toString(error);
+      if (
+        ["OperationNotSupported", "model_not_found", "rate_limit_exceeded"].includes(error.code) ||
+        error.type === "invalid_request_error"
+      ) {
+        console.log(`Handled error: ${errorStr}`);
+        continue;
+      }
+      console.warn(`Error in deployment ${deployment}: ${errorStr}`);
+      errors.push(errorStr);
+    }
+  }
+  if (errors.length > 0) {
+    throw new Error(`Errors list: ${errors.join("\n")}`);
+  }
+  assert.isNotEmpty(succeeded, "No deployments succeeded");
+  console.log(`Succeeded with (${succeeded.length}): ${succeeded.join(", ")}`);
+  return succeeded;
+}
+
+export async function sendRequestWithRecorder(
+  request: PipelineRequest,
+  recorder: Recorder
+): Promise<PipelineResponse> {
+  const client = createDefaultHttpClient();
+  const pipeline = createEmptyPipeline();
+  if (!isLiveMode()) {
+    for (const p of recorder.configureClientOptions({}).additionalPolicies ?? []) {
+      pipeline.addPolicy(p.policy);
+    }
+  }
+  return pipeline.sendRequest(client, request);
+}
+
+async function listOpenAIModels(cred: KeyCredential, recorder: Recorder): Promise<string[]> {
+  const request = createPipelineRequest({
+    url: "https://api.openai.com/v1/models",
+    headers: createHttpHeaders({
+      Authorization: cred.key,
+    }),
+    method: "GET",
+    timeout: 0,
+    withCredentials: false,
+    requestId: randomUUID(),
+  });
+  const response = await sendRequestWithRecorder(request, recorder);
+
+  const body = JSON.parse(response.bodyAsText as string);
+  const models = body.data
+    .map((model: { id: string }) => model.id)
+    .filter((id: string) => !id.match(/fine/));
```

Comment:
I did a recent update in openai tests to be more precise about what models to use, feel free to bring that in and of course it is not a blocker for this PR. I think we can start sharing things once the PR lands.